### ISSUE TITLE
Publishing a single photo

### ISSUE THREAD

**Issue by zotanmew** [LycheeOrg/Lychee#941]
When a single photo is set to public, but not the album it belongs to, the photo is not visible or viewable (with the link) publicly, even though the interface makes it seem that it should.

**Steps to reproduce the behavior:**
1. Change visibility to public on a photo belonging to a private album
2. Open the page while not logged in
3. Find that the photo is not visible

### Output of the diagnostics  [REQUIRED]
```
   Diagnostics
    -------
    Warning: Dropbox import not working. dropbox_key is empty.
    Warning: Pictures that are rotated lose their metadata! Please install Imagick to avoid that.



    System Information
    --------------
    Lychee Version (release):        4.2.2
    DB Version:                      4.2.2
    
    composer install:                --no-dev
    APP_ENV:                         production
    APP_DEBUG:                       false
    
    System:                          Linux
    PHP Version:                     7.4
    Max uploaded file size:          100M
    Max post size:                   100M
    MySQL Version:                   10.5.9-MariaDB
    
    Imagick:                         -
    Imagick Active:                  1
    Imagick Version:                 -
    GD Version:                      2.3.1
    
    Lychee total space:              1.05 GB
    Upload folder space:             1.01 GB
    System total space:              16.07 GB
    System free space:               10.77 GB (67%)



    Config Information
    --------------
    version:                         040202
    check_for_updates:               0
    sorting_Photos_col:              takestamp
    sorting_Photos_order:            ASC
    sorting_Albums_col:              max_takestamp
    sorting_Albums_order:            ASC
    imagick:                         1
    skip_duplicates:                 1
    small_max_width:                 0
    small_max_height:                360
    medium_max_width:                1920
    medium_max_height:               1080
    lang:                            en
    layout:                          1
    image_overlay:                   1
    image_overlay_type:              desc
    default_license:                 CC-BY-NC-SA-4.0
    compression_quality:             90
    full_photo:                      1
    delete_imported:                 0
    Mod_Frame:                       1
    Mod_Frame_refresh:               30
    thumb_2x:                        1
    small_2x:                        1
    medium_2x:                       1
    landing_page_enable:             0
    landing_owner:                   Laura Hausmann
    landing_title:                   Laura Hausmann
    landing_subtitle:                
    landing_facebook:                
    landing_flickr:                  
    landing_twitter:                 
    landing_instagram:               
    landing_youtube:                 
    landing_background:              
    site_title:                      Lychee v4
    site_copyright_enable:           1
    site_copyright_begin:            2020
    site_copyright_end:              2021
    additional_footer_text:          
    display_social_in_gallery:       0
    public_search:                   0
    SL_enable:                       0
    SL_for_admin:                    0
    public_recent:                   0
    recent_age:                      1
    public_starred:                  0
    downloadable:                    0
    photos_wraparound:               1
    map_display:                     0
    zip64:                           1
    map_display_public:              0
    map_provider:                    Wikimedia
    force_32bit_ids:                 0
    map_include_subalbums:           0
    update_check_every_days:         3
    has_exiftool:                    0
    share_button_visible:            0
    import_via_symlink:              1
    has_ffmpeg:                      1
    location_decoding:               0
    location_decoding_timeout:       30
    location_show:                   1
    location_show_public:            0
    rss_enable:                      0
    rss_recent_days:                 7
    rss_max_items:                   100
    prefer_available_xmp_metadata:   0
    editor_enabled:                  1
    lossless_optimization:           0
    swipe_tolerance_x:               150
    swipe_tolerance_y:               250
    local_takestamp_video_formats:   .avi|.mov
    log_max_num_line:                1000
    unlock_password_photos_with_url_param: 0
    nsfw_visible:                    1
    nsfw_blur:                       1
    nsfw_warning:                    0
    nsfw_warning_admin:              0
    map_display_direction:           1
    album_subtitle_type:             takedate
    upload_processing_limit:         12
```
### Browser and system
Chrome Stable v89.0.4389.82 on Arch Linux

**Comment by kamil4** [comment]
I agree that this is a little counterintuitive, but the way to view such photos in public mode is using the photo view link, which you can find under _Share Photo_ / _Direct Links_ (the top link in the dialog box that opens).

**Comment by zotanmew** [comment]
hm, okay. is partial sharing of albums something that is feasible, or should I close this issue?

**Comment by kamil4** [comment]
You can use the _Sharing_ screen to share private albums with individual users. Or do you mean something else? Can you describe what you mean by _partial sharing of albums_?

**Comment by zotanmew** [comment]
Oh, sorry for the confusion. What I mean is a setting that shows albums publicly that are not set to public (=all photos visible), but only with the photos that are set to public in them. Does that make sense?

**Comment by kamil4** [comment]
Thanks for explaining what you mean. Sorry, we don't have such a setting. The only way to see an album in public mode is for that album to be public. And once you make an album public, all photos inside are automatically public as well.

Photos individually made public are hidden by default. We recently merged a feature, however (see #932), which makes it possible to have them included in, e.g., tag albums -- perhaps that would be good enough for you? Note that that feature is not yet included in any of the releases, so if you want to experiment with it, you'll need to clone our git repository.

**Comment by zotanmew** [comment]
hm, okay. i bypassed the issue by copying the photos to a separate album, but i'd leave this issue open as i'd love to see this implemented at some point.
